### ISSUE TITLE
Using permissions.is_owner locks out all users, including the owner

### ISSUE THREAD

**Issue by acffordyce973** [Pycord-Development/pycord#1064]
### Summary

permissions.is_owner doesn't work even as the bot owner

### Reproduction Steps

Add `@permissions.is_owner()` to a slash command within a cog
Tried to use the slash command in Discord

### Minimal Reproducible Code

```python
@slash_command(default_permission=False)
@permissions.is_owner()
async def ping(self, ctx):
	"""Displays bot latency to Discord API."""
	await ctx.respond("Connection latency to Discord API is **" + str(round(self.bot.latency * 1000, 2)) + "** milliseconds!")
```

### Expected Results

The slash command should be greyed out to everyone except myself (the owner).

### Actual Results

The slash command is greyed out to everyone, including myself (the owner).

### Intents

discord.Intents().all()

### System Information

- Python v3.9.2-final
- py-cord v2.0.0-beta
    - py-cord pkg_resources: v2.0.0b4
- aiohttp v3.7.4.post0
- system info: Windows 10 10.0.22000

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

I cannot get `@permissions.is_owner()` to work for me on 2.0.0b4.
I've tried with and without `owner_id` in the `discord.Bot` constructor. If I output `bot.owner_id` then I get my ID fine (with and without it manually set).
Intents are set to `all()`.
Bot was invited with `scope=applications.commands%20bot`.
Tried with `default_permission=False` in `@slash_command` too.
Also tried `@slash_command(default_permission=False, permissions=permissions.is_owner())`
Tried both `discord.Bot` and `commands.Bot`
The command works fine when `@permissions.is_owner()` is removed.
Made another bot and tried that too.
It's been nearly 24 hours and the first bot has been restarted numerous times.

**Comment by krittick** [comment]
Can you show your imports? Are you by chance using the `discord.ext.commands` version of `is_owner()` instead of the one provided by `discord.commands.permissions` ?

**Comment by acffordyce973** [comment]
```
import discord
from discord.commands import slash_command, user_command, message_command, Option, permissions
from discord.ext import commands, tasks
```

**Comment by krittick** [comment]
I can't reproduce this with the provided code. For me, the command is only grayed out to non-owners (as expected). Can you share how you're initializing the bot (either your subclass or call to commands.Bot / discord.Bot)?

**Comment by acffordyce973** [comment]
I've found the cause and it was more or less user error.
I use `bot.owner_id` to save my ID to a config file and then load that in `commands.Bot`. It seems like I've managed to give it a `0` as an owner ID during the first run of each bot (as this is done before `bot.owner_id` exists) and it's just stuck with it. I've managed to fix it by making the command specific to a guild and then made it global again. I now have two `ping` commands but I'm hoping Discord will eventually get rid of the guild specific one.
